src/main/java/fi/aalto/cs/apluscourses/intellij/actions/APlusAuthenticationAction.java

```diff
@@ -69,11 +69,11 @@ public void update(@NotNull AnActionEvent e) {
   public void actionPerformed(@NotNull AnActionEvent e) {
     Project project = e.getProject();
     MainViewModel mainViewModel = mainViewModelProvider.getMainViewModel(project);
-    Course course = Optional.ofNullable(mainViewModel.courseViewModel.get())
-        .map(BaseViewModel::getModel)
-        .orElse(null);
-
-    if (course == null) {
+    Course course;
```

Comment:
perhaps this could work too? IMO improves the readability a bit... @OlliKiljunen

Reply:
Yes, this is better 👍